plugins/util.js

```diff
@@ -36,41 +52,195 @@ function mkdirp(directory) {
+              node {
+                name
+              }
+            }
+            categories {
+              edges {
+                node {
+                  name
+                }
+              }
+            }
           }
         }
       }
     }
   `;
 
-  const apolloClient = new ApolloClient({
-    uri: endpoint,
-    cache: new InMemoryCache(),
-  });
-
   let posts = [];
 
   try {
     const data = await apolloClient.query({ query });
-    posts = data.data.posts.edges.map(({ node = {} }) => node);
-    console.log(`[${process}] Successfully fetched posts from ${endpoint}`);
+    const nodes = [...data.data.posts.edges.map(({ node = {} }) => node)];
+
+    posts = nodes.map((post) => {
+      const data = { ...post };
+
+      if (data.author) {
+        data.author = data.author.node.name;
+      }
+
+      if (data.categories) {
+        data.categories = data.categories.edges.map(({ node }) => node.name);
+      }
+
+      return data;
+    });
+
+    console.log(`[${process}] Successfully fetched posts from ${apolloClient.link.options.uri}`);
   } catch (e) {
-    console.log(`[${process}] Failed to fetch posts from ${endpoint}: ${e}`);
+    console.log(`[${process}] Failed to fetch posts from ${apolloClient.link.options.uri}: ${e}`);
   } finally {
-    return posts;
+    return {
+      posts,
+    };
   }
 }
 
-module.exports.getAllPosts = getAllPosts;
+/**
+ * getSiteMetadata
+ */
+
+async function getSiteMetadata(apolloClient, process) {
+  const query = gql`
+    {
+      generalSettings {
+        description
+        language
+        title
+      }
+    }
+  `;
+
+  let metadata = {};
+
+  try {
+    const data = await apolloClient.query({ query });
+    metadata = { ...data.data.generalSettings };
+
+    if (!metadata.language || metadata.language === '') {
+      metadata.language = 'en';
+    } else {
+      metadata.language = metadata.language.split('_')[0];
+    }
+
+    console.log(`[${process}] Successfully fetched metadata from ${apolloClient.link.options.uri}`);
+  } catch (e) {
+    console.log(`[${process}] Failed to fetch metadata from ${apolloClient.link.options.uri}: ${e}`);
+  } finally {
+    return {
+      metadata,
+    };
+  }
+}
+
+/**
+ * getFeedData
+ */
+
+async function getFeedData(apolloClient, process) {
+  const metadata = await getSiteMetadata(apolloClient, process);
+  const posts = await getAllPosts(apolloClient, process);
+
+  return {
+    ...metadata,
+    ...posts,
+  };
+}
+
+/**
+ * generateFeed
+ */
+
+function generateFeed({ posts = [], metadata = {}, productionUrl }) {
+  const feed = new RSS({
+    title: metadata.title || '',
+    description: metadata.description,
+    site_url: productionUrl,
+    feed_url: `${productionUrl}/feed.xml`,
+    copyright: `${new Date().getFullYear()} ${metadata.title}`,
+    language: metadata.language,
+    pubDate: new Date(),
+  });
+
+  posts.map((post) => {
+    feed.item({
+      title: post.title,
+      guid: post.postId,
```

Comment:
i _think_ this needs to be something more specific. ive usually seen the guid simply the same value as the url

![image](https://user-images.githubusercontent.com/1045274/97244130-32540100-17ce-11eb-9c0c-f2f4e3383f3a.png)

Reply:
You are right. I will change to be the post URL 👍